src/main/kotlin/ecommerce/exception/GlobalExceptionHandler.kt

```diff
@@ -46,4 +47,83 @@ class GlobalExceptionHandler {
             )
         return ResponseEntity(errorMessage, HttpStatus.INTERNAL_SERVER_ERROR)
     }
+
+    @ExceptionHandler(MethodArgumentNotValidException::class)
+    fun handleValidationError(ex: MethodArgumentNotValidException): ResponseEntity<Map<String, String>> {
+        val errors =
+            ex.bindingResult.fieldErrors.associate {
+                it.field to (it.defaultMessage ?: "Invalid value")
+            }
+        return ResponseEntity.badRequest().body(errors)
+    }
+
+    @ExceptionHandler(ProductAlreadyInDBException::class)
+    fun handleProductAlreadyInDBException(ex: ProductAlreadyInDBException): ResponseEntity<ErrorMessageModel> {
+        val errorMessage =
+            ErrorMessageModel(
+                HttpStatus.CONFLICT.value(),
+                ex.message,
+            )
+        return ResponseEntity(errorMessage, HttpStatus.CONFLICT)
+    }
+
+    @ExceptionHandler(MemberEmailAlreadyExistsException::class)
+    fun handleEmailAlreadyExistsException(ex: MemberEmailAlreadyExistsException): ResponseEntity<ErrorMessageModel> {
+        val errorMessage =
+            ErrorMessageModel(
+                HttpStatus.CONFLICT.value(),
+                ex.message,
+            )
+        return ResponseEntity(errorMessage, HttpStatus.CONFLICT)
+    }
```

Comment:
```suggestion
    @ExceptionHandler(value = [
        ProductAlreadyInDBException::class,
        MemberEmailAlreadyExistsException::class,
    ])
    fun handleConflictException(ex: RuntimeException): ResponseEntity<ErrorMessageModel> {
        val errorMessage =
            ErrorMessageModel(
                HttpStatus.CONFLICT.value(),
                ex.message,
            )
        return ResponseEntity(errorMessage, HttpStatus.CONFLICT)
    }
```

We can handle multiple exceptions like this using a single method 👀

Reply:
This was a great tip 😃
I refactored the handler as suggested to group similar exceptions and reduce duplication, definitely cleaner now. Thanks!